pkg/minishift/network/settings.go

```diff
@@ -43,31 +57,263 @@ type NetworkSettings struct {
 	Gateway   string
 	DNS1      string
 	DNS2      string
+	UseDHCP   bool
+	Disabled  bool
+}
+
+// checkSupportForAddressAssignment returns true when the instance can support
+// minishift-set-ipaddress
+func checkSupportForAddressAssignment() bool {
+	if minishiftConfig.InstanceConfig.IsRHELBased &&
+		minishiftConfig.InstanceConfig.SupportsNetworkAssignment {
+		return true
+	} else {
+		atexit.ExitWithMessage(1, configureNetworkNotSupportedMessage)
+	}
+	return false
+}
+
+// This will return the address as used by libmachine
+func GetIP(driver drivers.Driver) (string, error) {
+	ip, err := driver.GetIP()
+	if err != nil {
+		return "", err
+	}
+	return ip, nil
+}
```

Comment:
I think it should be part of some utility package. we use GetIP in some parts of the code, it would be good if we can use one function for GetIP (for future).  May be in "github.com/minishift/minishift/cmd/minishift/cmd/util"

Reply:
This will be general IP/Network related stuff... this is not per se 'util' kind of functionality.
It is meant for overriding the driver values, but this will not be implemented at this stage...

If this ends up in util, it gets lost... JUST like the function `DetermineHostIP` did (peeps were unaware it existed)... I'd rather move this kind of functionality into a util.go file as part of the network package.